cg/models/downsample/downsample_meta_data.py

```diff
@@ -0,0 +1,167 @@
+            from_sample=self.original_sample.internal_id,
+            tumour=self.original_sample.is_tumour,
+            priority=Priority.standard,
+            customer=self.original_sample.customer,
+            application_version=application_version,
+        )
+        if self.sample_exists_in_statusdb(downsampled_sample.internal_id):
+            raise ValueError(f"Sample {downsampled_sample.internal_id} already exists in StatusDB.")
+        return downsampled_sample
+
+    def _generate_statusdb_downsampled_case(
+        self,
+    ) -> Family:
+        """Generate a case for the down sampled samples. The new case uses existing case data."""
+        downsampled_case: Family = self.config.status_db.add_case(
+            data_analysis=self.original_case.data_analysis,
+            data_delivery=self.original_case.data_delivery,
+            name=self.downsampled_case_name,
+            panels=["OMIM-AUTO"],
+            priority=self.original_case.priority,
+            ticket=self.original_case.latest_ticket,
+        )
+        downsampled_case.customer = self.original_case.customer
+        if self.case_exists_in_statusdb(downsampled_case.internal_id):
+            raise ValueError(f"Case {downsampled_case.internal_id} already exists in StatusDB.")
+        return downsampled_case
+
+    def case_exists_in_statusdb(self, case_internal_id: str) -> bool:
+        """Check if a case exists in StatusDB."""
+        case: Family = self.config.status_db.get_case_by_internal_id(case_internal_id)
+        if case:
+            return True
+        return False
+
+    def sample_exists_in_statusdb(self, sample_internal_id: str) -> bool:
+        """Check if a sample exists in StatusDB."""
+        sample: Sample = self.config.status_db.get_sample_by_internal_id(sample_internal_id)
+        if sample:
+            return True
+        return False
+
+    def has_enough_reads_to_downsample(self) -> bool:
+        """Check if the sample has enough reads to down sample."""
+        if not self.original_sample.reads > self.number_of_reads * 1_000_000:
+            raise ValueError(
+                f"Sample {self.original_sample.internal_id} does not have enough reads ({self.original_sample.reads}) to down sample to "
+                f"{self.number_of_reads}M."
+            )
+        return True
+
+    @property
+    def fastq_file_input_directory(self) -> Path:
+        """Get the latest version directory for a sample in housekeeper."""
+        return Path(
+            self.config.housekeeper_api.get_file_from_latest_version(
+                bundle_name=self.original_sample.internal_id, tags={SequencingFileTag.FASTQ}
+            ).path
+        ).parent
+
+    @property
+    def fastq_file_output_directory(self):
+        """Get the output directory for the down sampled sample."""
+        return Path("home", "proj", "production", "downsample", self.downsampled_sample.internal_id)
```

Comment:
I don't think this path should be hard coded. I think this should be part of cg.yalm in servers

Reply:
noted with a comment for now!